### DISCUSSION TITLE
Running Slinky with scala version 3

### DISCUSSION THREAD

**Discussion by olono1** [shadaj/slinky#598] [Q&A]
Hej there
we are doing a project with scala and we use scala version 3
**I wanted to ask you if slinky is compatible with scala version 3 and if anyone has tried it.** 
I am kind of new to the scala language and that is why I wanted to ask you if it is even possible to run slinky with scala version 3.

Thank you for your help!

**Comment by shadaj** [comment]
Slinky officially supports Scala 3!

**Reply by olono1** [reply]
May I ask if there are any tutorials or templates for running slinky with Scala 3?
We did search for them, but always ended up with errors that we could not resolve.
Thank you for your help.

**Reply by nigeleke** [reply]
A Scala3 example would be great. I used the HelloWorld `sbt new shadaj/create-react-scala-app.g8`, updated scalaVersion to 3.2.1, updated all the dependencies to their most recent versions, and have a compile time issue with the annotations package and the @react annotation itself.
```
[info] welcome to sbt 1.8.0 (Oracle Corporation Java 17.0.1)
[info] loading global plugins from ...\.sbt\1.0\plugins
[error] -- [E046] Cyclic Error: ...\src\main\scala\nigeleke\App.scala:18:1
[error] 18 |@react class App extends StatelessComponent {
[error]    | ^
[error]    | Cyclic reference involving val <import>
[error]    |
[error]    | longer explanation available when compiling with `-explain`
[error] -- [E008] Not Found Error: ...\src\main\scala\nigeleke\App.scala:4:19
[error] 4 |import slinky.core.annotations.react
[error]   |       ^^^^^^^^^^^^^^^^^^^^^^^
[error]   |       value annotations is not a member of slinky.core
[error] -- [E007] Type Mismatch Error: ...\src\main\scala\nigeleke\Main.scala:33:23
[error] 33 |    ReactDOM.render(App(), container)
[error]    |                    ^^^^^
[error]    |                    Found:    nigeleke.App
[error]    |                    Required: slinky.core.facade.ReactElement
```
